### PR TITLE
Pypi package

### PR DESCRIPTION
**Pull request by start974**
Add the file `setup.py` to distribute Alectryon packages and script on **Pypi**, and make it easier to install.
#41

### CODE REVIEW

**Comment by cpitclaudel** [comment]
Excellent job! 👏 thanks a lot, I wouldn't have had time to get this done.

I've pushed a few more commits to fix things up.  Can you squash your commits and mine (except the last one)?  After that I think we'll be good to go :)  I'll leave a few small comments inline.

File: .github/workflows/pypi.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install setuptools wheel twine
+    - name: Build and publish
+      env:
+        TWINE_USERNAME: ${{ secrets.PYPI_USERNAME }}
+        TWINE_PASSWORD: ${{ secrets.PYPI_PASSWORD }}
+      run: |
+        python setup.py sdist bdist_wheel
```
Comment (cpitclaudel) [inline .github/workflows/pypi.yml:25]: How does this compare to ` python3 -m build`?

File: setup.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    url='https://github.com/cpitclaudel/alectryon',
+    long_description_content_type='text/x-rst',
+    author=alectryon.__author__,
+    install_requires=REQUIRED,
+    packages=['alectryon'],
+    package_data={'':['assets/*']},
+    entry_points={'console_scripts': 'alectryon=alectryon.cli:main'},
+    python_requires='>=3.6',
+    license='MIT',
+)
```
Comment (cpitclaudel) [inline setup.py:31]: Could we merge this with setup.cfg?

Comment (start974) [reply setup.py:31]: I think too, the files can be merged.

File: .github/workflows/pypi.yml (hunk 1 above)
Comment (start974) [reply .github/workflows/pypi.yml:25]: Good question.
I'm not sure, but I think method with `python3 -m build` is more resent, but equivalent.

File: setup.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    long_description=LONG_DESCRIPTION,
+    url='https://github.com/cpitclaudel/alectryon',
+    long_description_content_type='text/x-rst',
+    author=alectryon.__author__,
+    install_requires=REQUIRED,
+    packages=['alectryon'],
+    package_data={'':['assets/*']},
+    entry_points={'console_scripts': 'alectryon=alectryon.cli:main'},
+    python_requires='>=3.6',
+    license='MIT',
```
Comment (start974) [inline setup.py:30]: @cpitclaudel 
Can you add keyword arguments:
- author
- author_email

File: .github/workflows/pypi.yml (hunk 1 above)
Comment (cpitclaudel) [reply .github/workflows/pypi.yml:25]: Cool, let's use that?

File: setup.py (hunk 2 above)
Comment (cpitclaudel) [reply setup.py:31]: Super.  Will you take care of that?

File: setup.py (hunk 3 above)
Comment (cpitclaudel) [reply setup.py:43]: Done :)

File: setup.py (hunk 2 above)
Comment (start974) [reply setup.py:44]: I've looked a little more in detail and it seems that it's not possible to put the content of the `setup.cfg` file in `setup.py`.
Also I saw that the techniques to make pipy package have well evolved, now there exists [poetry](https://python-poetry.org/) which allows to develop in clean environment(like [pipenv](https://pipenv.pypa.io/en/latest/)), build, publish...

**Comment by cpitclaudel** [comment]
I've done one last cleanup pass; can you have a look?  Then I'll merge :tada:

**Comment by start974** [comment]
I just hard-coded the attribute "author" this field doesn't support `attr` unfortunately, as said on the [setuptool](https://setuptools.readthedocs.io/en/latest/userguide/declarative_config.html) page.
You can look at all the information after installing the package by doing `pip show alectryon`.

For me everything is ok, you can merge.

File: .github/workflows/pypi.yml (hunk 1 above)
Comment (start974) [reply .github/workflows/pypi.yml:25]:
```suggestion
        make upload
```

File: Makefile [hunk 4]
```diff
@@ -2,5 +2,13 @@ test:
 	+$(MAKE) -C recipes clean
 	+$(MAKE) -C recipes
 
+.PHONY: dist
+
+dist:
+	python3 -m build
+
+upload:
```
Comment (start974) [inline Makefile:10]:
```suggestion
upload: dist
```

**Review by start974 — COMMENTED** [review]
I have no more change to propose, you can squash and merge.

**Comment by Zimmi48** [comment]
Cool! Do you intend to do a release soon?

**Comment by cpitclaudel** [comment]
Yes :) I just pushed two releases: one for 1.0 that was released a few months ago, and one fresh 1.1 release from `master`.  Thanks @start974!

**Pull request merged**